crates/astria-core/src/protocol/transaction/v1alpha1/action_group/mod.rs

```diff
@@ -0,0 +1,196 @@
+#[cfg(test)]
+mod tests;
+
+use std::fmt::{
+    self,
+    Debug,
+};
+
+use penumbra_ibc::IbcRelay;
+
+use super::{
+    action::{
+        ActionName,
+        BridgeLockAction,
+        BridgeSudoChangeAction,
+        BridgeUnlockAction,
+        FeeAssetChangeAction,
+        FeeChangeAction,
+        IbcRelayerChangeAction,
+        IbcSudoChangeAction,
+        Ics20Withdrawal,
+        InitBridgeAccountAction,
+        SequenceAction,
+        SudoAddressChangeAction,
+        TransferAction,
+        ValidatorUpdate,
+    },
+    Action,
+};
+
+trait BelongsToGroup {
+    const GROUP: ActionGroup;
+}
+
+macro_rules! impl_belong_to_group {
+    ($(($act:ty, $group:expr)),*$(,)?) => {
+        $(
+            impl BelongsToGroup for $act {
+                const GROUP: ActionGroup = $group;
+            }
+        )*
+    }
+}
+
+impl_belong_to_group!(
+    (SequenceAction, ActionGroup::General),
+    (TransferAction, ActionGroup::General),
+    (ValidatorUpdate, ActionGroup::General),
+    (SudoAddressChangeAction, ActionGroup::UnbundleableSudo),
+    (IbcRelayerChangeAction, ActionGroup::Sudo),
+    (Ics20Withdrawal, ActionGroup::General),
+    (InitBridgeAccountAction, ActionGroup::UnbundleableGeneral),
+    (BridgeLockAction, ActionGroup::General),
+    (BridgeUnlockAction, ActionGroup::General),
+    (BridgeSudoChangeAction, ActionGroup::UnbundleableGeneral),
+    (FeeChangeAction, ActionGroup::Sudo),
+    (FeeAssetChangeAction, ActionGroup::Sudo),
+    (IbcRelay, ActionGroup::General),
+    (IbcSudoChangeAction, ActionGroup::UnbundleableSudo),
+);
+
+impl Action {
+    const fn group(&self) -> ActionGroup {
+        match self {
+            Action::Sequence(_) => SequenceAction::GROUP,
+            Action::Transfer(_) => TransferAction::GROUP,
+            Action::ValidatorUpdate(_) => ValidatorUpdate::GROUP,
+            Action::SudoAddressChange(_) => SudoAddressChangeAction::GROUP,
+            Action::IbcRelayerChange(_) => IbcRelayerChangeAction::GROUP,
+            Action::Ics20Withdrawal(_) => Ics20Withdrawal::GROUP,
+            Action::InitBridgeAccount(_) => InitBridgeAccountAction::GROUP,
+            Action::BridgeLock(_) => BridgeLockAction::GROUP,
+            Action::BridgeUnlock(_) => BridgeUnlockAction::GROUP,
+            Action::BridgeSudoChange(_) => BridgeSudoChangeAction::GROUP,
+            Action::FeeChange(_) => FeeChangeAction::GROUP,
+            Action::FeeAssetChange(_) => FeeAssetChangeAction::GROUP,
+            Action::Ibc(_) => IbcRelay::GROUP,
+            Action::IbcSudoChange(_) => IbcSudoChangeAction::GROUP,
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub(super) enum ActionGroup {
+    General,
+    UnbundleableGeneral,
+    Sudo,
+    UnbundleableSudo,
```

Comment:
Bikeshed:

@noot left some comments on these where `Bundleable*` was removed and `Unbundleable*` introduced.

My preference would have been to go all out and use `General -> BundleableGeneral` and `Sudo -> BundleableSudo` (and keep `Unbundleable*`) so that this is always spelled out explicitly.

But I am going to leave the final call on this one to you.

Reply:
Changed to BundleableXXX. Agree it's easier to understand